yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/command/QueryConfig.java

```diff
@@ -146,6 +146,15 @@ void checkErrorInternal(@Nonnull SQLException e, @Nonnull String queryDescriptio
 
     }
 
+    /**
+     * Allow the config to control whether it should be executed or not.
+     * The executor will skip the execution and validation if the config returns false.
+     * @return true if the config should go on for execution, false if not.
+     */
+    boolean shouldExecute() {
```

Comment:
Note that this is similar (and can be converted to) the `SkipConfig` from #3049, once both Prs are merged in.

Reply:
Using `SkipConfig` is probably a good idea for this use-case as well. That way, we can get rid of `shouldExecute` altogether.  

We can probably  wait for #3049 to get in and then rebase this one on top of that.

Reply:
On the second though, I believe `SkipConfig` has slightly different semantics. If there is a `SkipConfig` we do not execute any of the other configs that are available for the query. More like a "break" statement. 

However, in this case, we need some `NoOpConfig` that is there but just not executed. It could be the same anonymous specialization of `QueryConfig` that we have for the case when a version is supported in #3049

Reply:
Agree. I can add that instead of the additional method to the interface if you think that's more appropriate. The one downside I can see for it is that changing the config type we will lose the content, but for now I think we can live t=with that.

Reply:
Done.